src/cocotb/handle.py

```diff
@@ -185,6 +185,42 @@ def __repr__(self) -> str:
         return type(self).__qualname__ + "(" + desc + ")"
 
 
+class RangeableObjectMixin(SimHandleBase):
+    """Base class for simulation objects that have a range."""
+
+    @cached_property
+    def range(self) -> Range:
+        """Return a :class:`~cocotb.types.Range` over the indexes of the array/vector."""
+        left, right = self._handle.get_range()
+
+        # guess direction based on length until we can get that from the GPI
+        length = self._handle.get_num_elems()
+        if length == 0:
```

Comment:
The need for making a distinction based on `len()` is not immediately clear to me (assume when we have an array instead of a vector?), a comment would help.